apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1428,13 +1428,25 @@ func (u *UniterAPI) WatchRelationUnits(args params.RelationUnits) (params.Relati
 func (u *UniterAPI) SetRelationStatus(args params.RelationStatusArgs) (params.ErrorResults, error) {
 	var statusResults params.ErrorResults
 
+	checker := u.st.LeadershipChecker()
+	token := checker.LeadershipCheck(u.unit.ApplicationName(), u.unit.Name())
```

Comment:
this token should be passed down to the SetStatus call. state.setStatusParams has room for a token.

Reply:
Trouble is SetStatus() is an interface method from the StatusSetter interface. So we'd need to do some refactoring. The application facade also doesn't use the token in the desired way. Could we keep as is and refactor holistically in another PR?

Reply:
I'm OK with that, since it's non critical. Please file a bug and leave a TODO(wallyworld).

FWIW, I don't think it's necessary to pass a token in the application facade (I assume you're referring to SetRelationsSuspended), because that's not being done by the application leader, but rather a client. And that code's meant to be going away anyway, right? In fact I think this PR is meant to be replacing it... except you're not setting the Joined status yet.

Reply:
Joined is only set by the uniter. When resuming a relation, the resume op sets the status to Joining.
The thing I'm referring to is setting the application status on behalf of the unit leader - it also uses the pattern here, ie grab and check a token before calling SetStatus

Reply:
> Joined is only set by the uniter. When resuming a relation, the resume op sets the status to Joining.

See SetRelationsSuspended in the application facade, specifically the TODO you left.

> The thing I'm referring to is setting the application status on behalf of the unit leader - it also uses the pattern here, ie grab and check a token before calling SetStatus
 Write

Ah, right. Yes, that's historical, and should also be fixed (later).

Reply:
There's no TODO left in SetRelationsSuspended
The facade code sets the status to Joining or Suspending as needed.